### PR TITLE
Test file copying preserves mtime

### PR DESCRIPTION
**Pull request by schneems**
Adds a test to ensure that a file copied via the AppCache feature preserves the file's original mtime. 

In #302 I determined that as long as the files copied into the cache dir preserve the original mtime then the FIFO cache cleaning logic that we rely on is valid. This test closes out this comment that asserted we need to verify whether or not the buildpack preserved mtime when copying files https://github.com/heroku/buildpacks-ruby/issues/302#issuecomment-2415128173.

#close 302

----
Commits are in historical order. I'm interested in feedback regarding tests, possibly some edge cases I've overlooked regarding test cases. app_cache.rs is approx 213 lines without comments or tests.

### CODE REVIEW

File: commons/src/cache/app_cache.rs [hunk 1]
```diff
@@ -448,4 +458,81 @@ mod tests {
...
+        let metadata = fs_err::metadata(store.cache.join("lol.txt")).unwrap();
+        let actual = filetime::FileTime::from_last_modification_time(&metadata);
+        assert_eq!(mtime, actual);
+
+        fs_err::remove_dir_all(&app_path).unwrap();
+        assert!(!store.path.join("lol.txt").exists());
+
+        store.load().unwrap();
+
+        let metadata = fs_err::metadata(store.cache.join("lol.txt")).unwrap();
```
Comment (schneems) [inline commons/src/cache/app_cache.rs:534]: Should be store.path and not store.cache

**Comment by schneems** [comment]
Cleaned up and re-opened https://github.com/heroku/buildpacks-ruby/pull/336
